### DISCUSSION TITLE
Boundary condition defined by a custom function depending on the last state and time

### DISCUSSION THREAD

**Discussion by Waschenbacher** [zwicker-group/py-pde#505] [Q&A] [answered]
Hello, I would like to simulate a convection-diffusion model for the following scenario:
![image](https://github.com/zwicker-group/py-pde/assets/116073761/390dd6ed-4585-4c20-a896-27414b4e2ffc)

- the velocity parameter $v$ of this convection-diffusion model depends on a custom function, which gets updated over time and depends on the last cell.
- the left boundary value should depends on both the last cell and the time. This dependence is defined by a custom function, which is probably not jitable.

Thanks to your help in several previous issues, it is now possible to add boundary defined by a non-jitable custom function which depends on the last cell. But for my ultimate goal, I tried several options and didn't succeed so far. Could you please provide me some further guidance?

**Comment by david-zwicker** [comment, marked as answer]
I think you best bet would be to write your own PDE class, like [in this example](https://github.com/zwicker-group/py-pde/blob/master/examples/pde_custom_class.py). This is not yet using numba compiled right-hand sides, but if your code is not really compilable by numba, this might be fine.

**Reply by Waschenbacher** [reply]
Thanks! I will give it a further try!
